campaigns/windows-10/logo-swap.html

```diff
@@ -0,0 +1,230 @@
+
+           var poll_retry = 0;
+           var check_default_timeout = undefined;
+           var is_firefox_default = undefined;
+           var cta_clicked = false;
+
+           isDefault(function yesDefault () {
+               sendMetric('impression-is-default');
+               showDefaultContent();
+               is_firefox_default = true;
+           }, function noDefault () {
+               sendMetric('impression-is-not-default');
+               showNonDefaultContent();
+               is_firefox_default = false;
+           });
+
+           var check_default_timeout = window.setInterval(function () {
+               if (poll_retry > 500) {
+                   clearInterval(check_default_timeout);
+               }
+               poll_retry++;
+               isDefault(function yesDefault () {
+                   //Default has switched while the snippet is opened
+                   if (is_firefox_default !== true) {
+                       showSuccessContent();
+                       is_firefox_default = true;
+                       if (cta_clicked === true) {
+                           sendMetric('conversion-is-default-changed-after-click');
+                       } else {
+                           sendMetric('conversion-is-default-changed-without-click');
+                       }
+                       clearInterval(check_default_timeout);
+                   }
+               }, function noDefault () {
+                   if (is_firefox_default !== false) {
+                       showNonDefaultContent();
+                       is_firefox_default = false;
+                   }
+               });
+           }, 1000);
+
+           function showDefaultContent () {
+               snippet_text_default.style.display = 'block';
+               snippet_text_success.style.display = 'none';
+               snippet_content_default.style.display = 'block';
+               snippet_content_not_default.style.display = 'none';
+               snippet_icon.src = '{{ icon }}';
+
+               for (var i=0, k=buttons.length; i<k; i++) {
+                   buttons[i].addEventListener('click', function (e) {
+                       e.preventDefault();
+                       sendMetric('conversion-is-default-click', function () {
+                           window.location = '{{ url_default|safe }}';
+                       });
+                   }, false);
+               }
+           }
+
+           function showSuccessContent () {
+               snippet_text_default.style.display = 'none';
+               snippet_text_success.style.display = 'block';
+               snippet_content_default.style.display = 'block';
+               snippet_content_not_default.style.display = 'none';
+               snippet_icon.src = '{{ icon }}';
+
+               for (var i=0; i < buttons.length; i++) {
+                   buttons[i].addEventListener('click', function (e) {
```

Comment:
So adding an event listener on `click` to send metrics is problematic since we added a global event listener for clicks.This results in sending a total of 3 metrics per click
1. `conversion-is-default-click`
2. then the global event listener catches the event and sends `click`
3. and then it emulates `click()` to the DOM element that was actually clicked which resends `conversion-is-default-click`.

To simplify the things a bit and solve solve the multiple metrics issue I changed the global click handler to support custom metric names. https://github.com/mozilla/snippets-service/commit/d06d621a0282cce0ecf7d084b878df8e0dfa1ab1

From now on if a clicked `<a>` element has a `data-metric` attribute then the global listener will send this value instead of `click`. If it doesn't it defaults to `click`.

This code should be changed to set the data-metric attribute
```javascript
for (var i=0; i < buttons.length; i++) {
  buttons[i].dataset.metric = 'conversion-is-default-click';
}
```

and let the global click handler to do the rest. You don't have to handle `click` events yourself.

If you do need to capture `click` events for other reasons, remember do not directly trigger `sendMetric` but just set the data attribute.